packages/web/src/scss/tools/_utilities.scss

```diff
@@ -1,9 +1,37 @@
 // Inspired by React UI and Bootstrap
+@use '@tokens' as tokens;
 @use 'sass:list';
 @use 'sass:map';
 @use 'sass:meta';
+@use 'sass:string' as sass-string;
 @use 'string';
 
+@function generate-token-map($tokens-key: null, $prefix: null, $replacement: null, $scope: null) {
+    $result: ();
+    $tokens: if(
+        map.has-key(meta.module-variables(tokens), $tokens-key),
+        map.get(meta.module-variables(tokens), $tokens-key),
+        null
+    );
+
+    @each $key, $values in $tokens {
+        @each $subkey, $value in $values {
+            // Check if the scope is a list, the subkey is not null, and the subkey exists in the scope list.
+            @if meta.type-of($scope) == 'list' and $subkey != null and list.index($scope, $subkey) != null {
+                // Construct the new key by combining the prefix, formatted key, and modified subkey.
+                $name: $prefix +
+                    if(meta.type-of($key) == 'number' and $key < 10, '0' + $key, $key) +
```

Comment:
The number check also needs a comment, please.

Reply:
I think we mostly use comments in the function description or number references at the top of a file.